File: String/1678. Goal Parser Interpretation/interpret.java
class Solution {
    public String interpret(String command) {
        String result = "";
        boolean bracket = false;
        String temp = "";
        for (char ch : command.toCharArray()) {
            if (ch == '(') {
                bracket = true;
            } else if (ch == ')') {
                if (temp.equals("")) {
                    result += 'o';
                } else {
                    result += temp;
                    temp = "";
                }
                bracket = false;
            } else if (bracket) {
                temp += ch;
            } else {
                result += ch;
            }
        }
        return result;
    }
}

File: String/1678. Goal Parser Interpretation/interpret.py
class Solution:
    def interpret(self, command: str) -> str:
        result = ""
        bracket = False
        temp = ""
        for ch in command:
            if ch == '(':
                bracket = True
            elif ch == ')':
                if temp == "":
                    result += 'o'
                else:
                    result += temp
                    temp = ""
                bracket = False
            elif bracket:
                temp += ch
            else:
                result += ch
        return result